### ISSUE TITLE
Source????? Is what node and how?

### ISSUE THREAD

**Issue by ewebgh33** [jitcoder/lora-info#18]
Hi

WHAT is this "source" node you have in your screenshot?

I found a reddit post saying it's actuall RGThree's "display any" node, but connecting one of those to _trigger_words_ and _example_prompt_, does not populate the RG3-displayany node, with the information. The nodes just sit there, displaying, nothing. 

I tried loading different loras, and the text in the main lorainfo node changes (trigger and prompt are buried in other info, there's a LOT of info). BUT nothing ever populates through to the source node, when attemping to use RG3 display-any.

If it's NOT the RG3 displayany, what is it please? I have a lot of custom nodes, and there is nothing called "source". So it must be something you've connected but then renamed yourself?
Renaming nodes is fine for personal work, but not for sharing flows. There are so many nodes and node authors than randomly renaming things leaves no real way to trace back what node is what.

TL;DR, what is the source node, and why can't I get anything to connect to those two outputs and display the trigger word and example prompt?

Thanks

**Comment by jitcoder** [comment]
hello

> WHAT is this "source" node you have in your screenshot?
this screenshot its from a while ago, the node used was RGThree's display any node. I did not rename anything. You can connect the outputs to any node that takes in `STRING` or `TEXT` inputs.

I was recently addressing a feature request and just googled "comfyui display text node"
<img width="919" alt="image" src="https://github.com/user-attachments/assets/288596dc-4f24-4a74-ad72-35be4f7c2520">

I installed that node and connected it.

<img width="1435" alt="image" src="https://github.com/user-attachments/assets/9825313b-64b5-40e5-876c-089a9ab1895d">

> I tried loading different loras, and the text in the main lorainfo node changes (trigger and prompt are buried in other info, there's a LOT of info). BUT nothing ever populates through to the source node, when attemping to use RG3 display-any.

There was a reddit thread I came across where someone was also asking this exact thing, turns out they never actually queued the prompt. The outputs and inputs in comfyui "execute" when a prompt is queued. ie. the node's output isn't actually picked up by comfyui until you queue a prompt. When you are selecting LoRAs in the dropdown that 'trigger' is sent to the node and the node is able to update itself, but it cannot interact with other nodes (ie. send its outputs or get inputs) until a prompt is queued.

> If it's NOT the RG3 displayany, what is it please? I have a lot of custom nodes, and there is nothing called "source". So it must be something you've connected but then renamed yourself?
Renaming nodes is fine for personal work, but not for sharing flows. There are so many nodes and node authors than randomly renaming things leaves no real way to trace back what node is what.

I didn't rename anything.. it was rgthrees display any node, but it doesn't have to be the display any node. you can use any node you want that takes text/string as an input.

Hopefully this information helps. let me know if you are still encountering issues (re-open the issue).

**Comment by jitcoder** [comment]
I'm going to add a FAQ section in the readme, hopefully it will avoid confusion for other people

**Comment by ewebgh33** [comment]
Thanks, an FAQ would have been helpful.

It wasn't clear that the outputs only populate when queued. To me, that's not so helpful. So queue twice - once to get the trigger word out, then paste trigger word into prompt, 2nd queue to actually try get the image generation I want. I guess this is just how comfy works. So thanks for the sarcastic google screenshot, but there was never an indication that text would only be displayed after queue, aside from... one reddit post? Cool cool.

Seems faster to just scan the metadata with my own eyeballs, find the trigger myself, copy paste it, rather than trying to get it out into its own box for easier copying. I suppose then the intended use of those ports is something else, like putting lora info into image metadata, whereas I assumed it was to pull out the trigger words easily and make life easier.

**Comment by jitcoder** [comment]
I've updated the readme with an FAQ know so that it helps others as well.

> So queue twice - once to get the trigger word out, then paste trigger word into prompt, 2nd queue to actually try get the image generation I want.

You shouldn't have to queue it twice. For example if you wanted to generate an image using the example_prompt 

I've recorded an example usage
![lorainfo_example](https://github.com/user-attachments/assets/8272bf3d-9003-43ae-bf2d-8403ce51e1cd)


You can do more advanced things (by using string manipulation nodes) that can take a prompt that you write, and concatenate trigger_words into your prompt before sending it to a clip encoder.

My personal usage goal for this node was so that I wouldn't have to type or copy/paste anything while trying out different loras.

let me know if this sorts you out. if need be, I'm more than happy to hop on a zoom with you to help you accomplish your goal.

**Comment by ewebgh33** [comment]
Thanks, that's a great help.
Appreciate it! :)
